Helper/Order.php

```diff
@@ -701,14 +688,60 @@ public function formatReferenceUrl($reference)
     }
 
     /**
-     * Format the (class internal) unambiguous transaction state out of the type and status
+     * Format the (class internal) unambiguous transaction state out of the type and status.
+     * Infer eventually missing states. Represent Bolt partial capture AUTHORIZED state as CAPTURED.
      *
      * @param \stdClass $transaction
+     * @param OrderPaymentInterface $payment
      * @return string
      */
-    public function getTransactionState($transaction)
+    public function getTransactionState($transaction, $payment)
     {
-        return $transaction->type.":".$transaction->status;
+        $transactionState = $transaction->type.":".$transaction->status;
+        $prevTransactionState = $payment->getAdditionalInformation('transaction_state');
+        $prevCaptures = $payment->getAdditionalInformation('captures') ?: [];
+
+        if (!$prevTransactionState) {
+            if (in_array($transactionState, [self::TS_ZERO_AMOUNT, self::TS_CREDIT_COMPLETED])) {
+                return $transactionState;
+            }
+            return self::TS_PENDING;
```

Comment:
how will that work for old transaction that is completed and now merchant want to refund?

Reply:
@daisy1754
I have just refunded some transactions completed before this code update. And a few even older ones, from about 2 moths ago. Probably it would fail on some real old ones, or created with old plugin version. We should deal with them manually if / when it happens.